### PR TITLE
Suggestion: impl OperationOutput for Rejections.

### PR DESCRIPTION
**Pull request by yassun4dev**
It would be meaningful for the API Client to output error responses (Status, ResponseBody) to OpenAPI for error handling.

As a simple example, I would implement OperationOutput to "JsonRejection".

But maybe I should work on the axum side instead of the aide.
It would be better to ask JsonRejection itself to tell what error codes it can return.
In that case, the modification of [composite_rejection!](https://github.com/tokio-rs/axum/blob/e0ef641e5fc12048b1632cf2feebddac19a2f8ad/axum/src/extract/rejection.rs#L123).

What do you think of this idea?

### CODE REVIEW

**Comment by tamasfe** [comment]
Thanks! I definitely agree with this. Anything that could appear in a response should implement OperationOutput.

Although I have a feeling that at some point we should make these opt-in, e.g. malformed JSON errors aren't really useful for an API documentation. It's just a side note, we can always make code generation more customizable later.

> But maybe I should work on the axum side instead of the aide.

Moving implementation to the framework side would definitely allow more flexibility and less burden here, but aide is nowhere near being popular enough nor is officially blessed by axum so I don't see that happening.

File: crates/aide/src/axum/outputs.rs [hunk 1]
```diff
@@ -122,3 +127,31 @@ impl<T> OperationOutput for Html<T> {
         }
     }
 }
+
+impl OperationOutput for JsonRejection {
+    type Inner = Self;
+
+    fn operation_response(ctx: &mut GenContext, operation: &mut Operation) -> Option<Response> {
+        Json::<String>::operation_response(ctx, operation)
```
Comment (tamasfe) [inline crates/aide/src/axum/outputs.rs:135]: This means that the response is a JSON string, e.g. `"error happened"` with `application/json` content type, is that correct?

**Review by tamasfe — CHANGES_REQUESTED** [review]

**Comment by tamasfe** [comment]
> But, I think this decision of "Which errors to output to OpenAPI" is highly personal.

Indeed, I would probably make it customizable via the context, with something like `all_error_responses`. At the same time I think that a few errors presented in this PR are not going to hurt anyone, so we can leave it as-is and get back to it later if needed, up to you.

> Should I implement `JsonRejection::inferred_responses` and let the user choices whether to use the value in `Json::inferred_responses` with `features` or `GEN_CTX`?

I would prefer the runtime context for this, I don't like tying behaviour to features because you cannot opt-out of them, and you cannot mix them. It would make generating documentation with all error codes and without all error codes within the same application impossible.

File: crates/aide/src/axum/outputs.rs (hunk 1 above)
Comment (yas7010uv) [reply crates/aide/src/axum/outputs.rs]: Thank you!!

I fixed.

**Comment by tamasfe** [comment]
Yes, looks good to me!

**Comment by yassun4dev** [comment]
I added OperationOutput to FormRejection.

I tried to do the same thing for OperationInput, but there is no place to define `inferred_responses`.
There needs to define `inferred_responses` method.

I think I will end the PR with this suggestion.

**Comment by tamasfe** [comment]
> I tried to do the same thing for OperationInput, but there is no place to define inferred_responses.
There needs to define inferred_responses method.

Makes sense, I'm preparing to update the axum dependency to `0.6.0`. I'll add `inferred_responses` to inputs as well in the next version.

I'll merge this in the meantime, thank you!

**Pull request merged**